docs/architecture/adr-076-combine-spec-repo.md

```diff
@@ -0,0 +1,97 @@
+# ADR 076: Combine Spec and Tendermint Repositories 
+
+## Changelog
+
+- 2022-02-04: Initial Draft. (@tychoish)
+
+## Status
+
+Accepted.
+
+## Context
+
+While the specification for tendermint was original in the same
+repository as the Go implementation, at some point the specification
+was split from the core tendermint repository and maintained
+separately from the implementation. While this makes sense in
+promoting a conceptual separation of specification and implementation,
+in practice this separation was a premature optimization apparently
+aimed at supporting alternate implementations of tendermint. 
+
+However, maintaining a spec repo has ended up being a premature
+optimization: there are no current projects to develop alternate
+implementations of tendermint based on the common specification, and
+having separate repositories creates an ongoing maintenance burden to
+support versioning and publication.
+
+## Decision
+
+The specification repository should be merged
+
+Stakeholders including representatives from the maintainers of the
+spec, the go implementation, and the tendermint rust library, agreed
+to merge the repositories in the tendermint core dev meeting on 27
+January 2022.
+
+## Alternative Approaches
+
+The main alternative to this was *do nothing* and continue to maintain
+separate versioning for both repositories, complicating the release
+process for both repositories and creating a difficult to resolve 
+
+## Detailed Design
+
+Within the `tendermint` repository, do the following to merge the
+repositories keeping history: 
+
+```bash
+git remote add spec git@github.com:tendermint/spec.git
+git fetch spec
+git checkout -b spec-master spec/master
+mkdir spec
+git ls-tree -z --name-only HEAD | xargs -0 -I {} git mv {} subdir/
+git commit -m "spec: organize specification prior to merge"
+git checkout -b spec-merge-mainline origin/master
+git merge --allow-unrelated-histories spec-master
+```
```

Comment:
Suggestion to summarize what the script is doing at a high level.

```suggestion
Clone and merge the master branch of the `tendermint/spec` repository
as a branch of the `tendermint/tendermint`, to ensure the commit history
of both repositories remains intact.


### Implementation Instructions

1. Within the `tendermint` repository, execute the following commands 
   to add a new branch with the history of the master branch of `spec`:

```bash
git remote add spec git@github.com:tendermint/spec.git
git fetch spec
git checkout -b spec-master spec/master
mkdir spec
git ls-tree -z --name-only HEAD | xargs -0 -I {} git mv {} subdir/
git commit -m "spec: organize specification prior to merge"
git checkout -b spec-merge-mainline origin/master
git merge --allow-unrelated-histories spec-master
```
```